src/Test/Plutip/Internal/LocalCluster.hs

```diff
@@ -128,22 +130,25 @@ withPlutusInterface :: forall (a :: Type). PlutipConfig -> (ClusterEnv -> IO a)
 withPlutusInterface conf action = do
   -- current setup requires `cardano-node` and `cardano-cli` as external processes
   checkProcessesAvailable ["cardano-node", "cardano-cli"]
-
   withLocalClusterSetup conf $ \dir clusterLogs _walletLogs nodeConfigLogHdl -> do
     result <- withLoggingNamed "cluster" clusterLogs $ \(_, (_, trCluster)) -> do
       let tr' = contramap MsgCluster $ trMessageText trCluster
       clusterCfg <- localClusterConfigWithExtraConf (extraConfig conf)
       withRedirectedStdoutHdl nodeConfigLogHdl $ \restoreStdout ->
-        withCluster tr' dir clusterCfg mempty $ \rn -> do
-          restoreStdout $ runActionWthSetup rn dir trCluster action
+        retryClusterFailedStartup $
+          withCluster tr' dir clusterCfg mempty $ \rn -> do
+            restoreStdout $ runActionWthSetup rn dir trCluster action
     handleLogs dir conf
     return result
   where
-    runActionWthSetup rn dir trCluster userActon = do
+    runActionWthSetup rn dir trCluster userAction = do
       let tracer' = trMessageText trCluster
       waitForRelayNode tracer' rn
-      maybePort <- handleChainIndexLaunch (chainIndexMode conf) rn dir
-      let cEnv =
+      mChainStarted <- handleChainIndexLaunch (chainIndexMode conf) rn dir
```

Comment:
An IMHO, but this stuff starting to get a bit involved. I think because of the way `handleChainIndexLaunch` is done.

Maybe do something like
```haskell
getChIndexPort :: ChainIndexMode -> Maybe ChainIndexPort
```

and then clearly branch on it like
```haskell
mPort = getChIndexPort (chainIndexMode conf)
case mPort of
  Just port -> bracket (startIndex rn dir) stopIndex (<make cEnv and run setup and actions>)
  Nothing ->  (<make cEnv and run setup and actions>)
```
or something like that, and trash `handleChainIndexLaunch`